lib/util/extract-sourcemap.js

```diff
@@ -0,0 +1,18 @@
+let { readFileSync } = require("fs");
+let regex = /sourceMappingURL=data:application\/json;base64,(\S+)/;
+
+// Extract the given attributes of a sourcemap as JSON from a given source file
+//
+// Exported as a global so it can be used with node -r
+global.extractSourcemap = (fileName, attributes) => {
```

Comment:
This is ugly... But I don't know any other solution. Suggestions welcome.

Reply:
This could also be a script taking the filename and attributes as arguments, like

    let [fileName, ...attributes] = process.argv.slice(2);

Reply:
I thought about that as well... But then it is an executable and shouldn't be in the lib folder, right? What about this?

```js
let extractSourcemap = (...) => { ... };

if(require.main === module) {
    let [fileName, ...attributes] = process.argv.slice(2);
    console.log(extractSourcemap(fileName, attributes));
} else {
    module.exports = extractSourcemap;
}
```

Reply:
TIL `require.main` 🐍 👍 

However, perhaps that's unnecessarily complicated. Elsewhere this works:

```
$ node -r fs -p "fs.realpathSync(process.argv[1]);" "/tmp/../etc"
/etc
```

So there `-r fs` actually makes that module accessible as a local variable `fs`. However, the same doesn't seem to work for more complex module paths:

```
$ node -r "faucet-pipeline/lib/util/runner" -p runner
[eval]:1
runner
^

ReferenceError: runner is not defined
```

If we can figure out how that works, we can invoke the function at will from the outside, which seems cleaner.

Update: Huh, it seems all stdlib modules are actually hidden properties of `global`, without them even being `require`d. I'm very confused right now.

Reply:
🙄 le sigh...

In that case, your `require.main` switch seems appropriate. I'm not entirely sure whether I'd prefer a global variable (your original approach) or the module acting as a CLI script (@mkhl's suggestion) in that case, so feel free to choose whatever you think is best.

FWIW, I believe you don't need `else` either way:

```
module.exports = extractSourcemap;

if(require.main === module) {
    // CLI / global
}
```

Reply:
* Ok, let's go with executing the file, Python-Style.
* Can't we go with `node "$root/../node_modules/faucet-pipeline/lib/util/extract-sourcemap"`? Why do this hole resolve dance?
* extract-sourcemap: Any suggestions for a name? I'm not sure how this can be expressed in a single word... I mean... this thing extracts a source map from source code.

Reply:
I can also just make that function its own NPM module with executable and give it a silly name...